### PR TITLE
chore: Follow rust naming convention

### PR DESCRIPTION
**Pull request by 0xYYY**
# Related issue(s)

Related to https://github.com/noir-lang/noir/issues/780.

# Description

## Summary of changes

Follow the [Rust API Gudeline](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case).

Modified cases:
`IRGenerator`, `SSAFunction` following the `UUID` -> `Uuid` example.
`LValue`, `LValueRhs`, `HirLValue` following the `USize` -> `Usize` example.

Please let me know if any of these seems inappropriate.

## Dependency additions / changes
N/A

## Test additions / changes
N/A

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

**Comment by TomAFrench** [comment]
@0xYYY Could you run `cargo fmt` on this and push again please?

**Comment by kevaundray** [comment]
Strange, is it really `Lvalue` and not `LValue` ?

**Comment by 0xYYY** [comment]
> Strange, is it really `Lvalue` and not `LValue` ?

I'm actually not totally sure about this one. So would like your inputs on this.

The guideline states that the usage of `Usize` instead of `USize`, but as you pointed out, Rust itself uses `LValue`. Perhaps we can leave this case out of this PR first if it's not that clear which should be used.

**Comment by kevaundray** [comment]
> > Strange, is it really `Lvalue` and not `LValue` ?
> 
> I'm actually not totally sure about this one. So would like your inputs on this.
> 
> The guideline states that the usage of `Usize` instead of `USize`, but as you pointed out, Rust itself uses `LValue`. Perhaps we can leave this case out of this PR first if it's not that clear which should be used.

Yeah , could we leave this one out please

File: crates/noirc_evaluator/src/ssa/context.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use crate::ssa::{
     acir_gen::Acir,
     block::{BasicBlock, BlockId},
     conditional::{DecisionTree, TreeBuilder},
-    function::{FuncIndex, SSAFunction},
+    function::{FuncIndex, SsaFunction},
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa/context.rs:6]: good catch

**Review by kevaundray — APPROVED** [review]

**Comment by kevaundray** [comment]
Thanks for resolving the conflicts!

**Review by kevaundray — APPROVED** [review]

**Pull request merged**